src/test/java/net/snowflake/ingest/streaming/internal/OAuthTest.java

```diff
@@ -0,0 +1,107 @@
+package net.snowflake.ingest.streaming.internal;
+
+import java.sql.Connection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+import net.snowflake.ingest.TestUtils;
+import net.snowflake.ingest.connection.OAuthCredential;
+import net.snowflake.ingest.streaming.*;
+import net.snowflake.ingest.utils.Constants;
+import org.junit.Before;
+import org.junit.Test;
+
+public class OAuthTest {
```

Comment:
Since OAuth integration is impossible to perform without backdoor function. Most unit test would be in `OAuthManagerTest`. All integration tests will be added in dew later.

Reply:
Sounds good